server/src/main/java/com/cloud/vm/snapshot/VMSnapshotManagerImpl.java

```diff
@@ -1045,16 +1037,13 @@ protected VMSnapshot retrieve() {
         long vmId;
 
         public VmJobVirtualMachineOutcome(final AsyncJob job, final long vmId) {
-            super(VirtualMachine.class, job, VmJobCheckInterval.value(), new Predicate() {
-                @Override
-                public boolean checkCondition() {
-                    AsyncJobVO jobVo = _entityMgr.findById(AsyncJobVO.class, job.getId());
-                    assert (jobVo != null);
-                    if (jobVo == null || jobVo.getStatus() != JobInfo.Status.IN_PROGRESS)
-                        return true;
+            super(VirtualMachine.class, job, VirtualMachineManager.VmJobCheckInterval.value(), () -> {
+                AsyncJobVO jobVo = _entityMgr.findById(AsyncJobVO.class, job.getId());
+                assert (jobVo != null);
+                if (jobVo == null || jobVo.getStatus() != JobInfo.Status.IN_PROGRESS)
```

Comment:
instead of using a conditional like "if true", return true, otherwise, return false; what if we used `return jobVo == null || jobVo.getStatus() != JobInfo.Status.IN_PROGRESS`